pulp_2to3_migration/app/plugin/iso/repository.py

```diff
@@ -72,3 +72,41 @@ def migrate_to_pulp3(cls, pulp2distributor, repo_version):
         distribution, created = FileDistribution.objects.update_or_create(**base_config)
 
         return publication, distribution, created
+
+    @classmethod
+    def needs_new_publication(cls, pulp2distributor):
+        """
+        Check if a publication associated with the pre_migrated distributor needs to be recreated.
+
+        Nothing in a Pulp 2 distributor configuration can cause a Pulp3 publication to change.
+
+        Args:
+            pulp2distributor(Pulp2Distributor): Pre-migrated pulp2 distributor to check
+
+        Return:
+            bool: True, if a publication needs to be recreated; False if no changes are needed
+        """
+        return False
+
+    @classmethod
+    def needs_new_distribution(cls, pulp2distributor):
+        """
+        Check if a distribution associated with the pre_migrated distributor needs to be recreated.
+
+        Args:
+            pulp2distributor(Pulp2Distributor): Pre-migrated pulp2 distributor to check
+
+        Return:
+            bool: True, if a distribution needs to be recreated; False if no changes are needed
+
+        """
+        if not pulp2distributor.pulp3_distribution:
```

Comment:
Right now it looks like this code is duplicated in 3 places. Can we make it the default implementation?

Reply:
It's different in for the docker, because it has `repo-registry-id` instead of `relative_url`.

I tried to avoid default implementation on purpose, so there are no implications when new plugin is added, e.g. pulp_deb. The problem is that if the default logic is wrong for the plugin and a plugin writer is unaware of these methods and everything just works, distribution or publication might not be recreated when they potentially should be. So throwing the ImplementationError seemed to me like a good way to ensure plugin writers are conscious of what logic is applied here.

What do you think? I can be convinced that it's not the best way to handle that.

Reply:
```ImplementationError``` sounds fine and is probably the best thing to do if we can't have a default.  I'm on the fence whether we should just have duplicate code or try to make it into a utility function kind of thing -- but this is fine with me if you have a preference for this way.

Reply:
No preference, utility function sounds like a fine idea.